### PR TITLE
CLEAN: use preferred pattern for index accessors

### PR DESCRIPTION
**Pull request by Dr-Irv**
pyright 1.1.271 had a bug https://github.com/microsoft/pyright/issues/3955 with my preferred way of handling some accessors, that was fixed in 1.1.272, so this PR uses the preferred implementation.

### CODE REVIEW

**Comment by bashtage** [comment]
Does this need tests, or do some already exist?

**Comment by Dr-Irv** [comment]
> Does this need tests, or do some already exist?

The tests already exist.  What this change does is make it so we have one set of declarations for `round`, `ceil`, etc., whereas I had to do two before because of the pyright bug.

File: pandas-stubs/core/indexes/accessors.pyi [hunk 1]
```diff
@@ -157,7 +157,7 @@ _DTRoundingMethodReturnType = TypeVar(
     TimedeltaSeries,
     TimestampSeries,
     DatetimeIndex,
-    # TimedeltaIndex
```
Comment (twoertwein) [inline pandas-stubs/core/indexes/accessors.pyi:160]: looks like a leftover

**Review by twoertwein — APPROVED** [review]
Looks good to me - glad to re-use the definitions from `_DatetimeRoundingMethods[TimedeltaIndex]`!

File: pandas-stubs/core/indexes/accessors.pyi (hunk 1 above)
Comment (Dr-Irv) [reply pandas-stubs/core/indexes/accessors.pyi:160]: It was a leftover from the "right" way to do it - that's what is put back in this PR!

Comment (twoertwein) [reply pandas-stubs/core/indexes/accessors.pyi:160]: oh, stupid me :)

**Comment by twoertwein** [comment]
Thanks @Dr-Irv !

**Pull request merged**